conntable.go

```diff
@@ -63,30 +67,60 @@ func (t *connTable) Register(ck ckey, matchedRule *Rule, srcIP, srcPort string,
 	if _, ok := t.table[ck]; ok {
 		// TODO: wut?
 	} else {
+
 		logger.Debugf("[contable] registering %s:%s->%d", srcIP, srcPort, targetPort)
 
 		t.table[ck] = &Metadata{
-			Added:      time.Now(),
-			Rule:       matchedRule,
-			TargetPort: targetPort,
+			Added:       time.Now(),
+			LastUpdated: time.Now(),
+			Rule:        matchedRule,
+			TargetPort:  targetPort,
 			//TargetIP:   targetIP,
 		}
+
 	}
 }
 
-func (t *connTable) FlushOlderThan(s time.Duration) {
+func (t *connTable) FlushOlderOnes() {
 	t.mtx.Lock()
 	defer t.mtx.Unlock()
 
-	threshold := time.Now().Add(-1 * s)
+	now := time.Now()
+
+	minutes := 30
+
+	for minutes >= 0 {
 
-	for ck, md := range t.table {
-		if md.Added.Before(threshold) {
-			delete(t.table, ck)
+		duration := time.Duration(minutes) * time.Minute
+		threshold := now.Add(-1 * duration)
+
+		for ck, md := range t.table {
+			if md.LastUpdated.Before(threshold) {
+				delete(t.table, ck)
+			}
+		}
+
+		if len(t.table) < t.softLimit {
+			break
+		} else {
+			minutes -= 10
 		}
 	}
 }
 
+func (t *connTable) updatePacketTime(ck ckey) {
+	t.mtx.RLock()
+	defer t.mtx.Unlock()
```

Comment:
The travis build is not showing as broken (see #20). Until that is fixed, make sure you look at the entire log: https://travis-ci.org/kung-foo/freki/builds/217986824#L520